include/libtorrent/session.hpp

```diff
@@ -131,6 +121,44 @@ namespace libtorrent
 		boost::shared_ptr<aux::session_impl> m_impl;
 	};
 
+	struct TORRENT_EXPORT session_params
+	{
+		session_params(settings_pack sp = settings_pack()
+			, int flags = default_flags)
+			: settings(sp)
+			, start_flags(flags)
+#ifndef TORRENT_DISABLE_DHT
```

Comment:
Do you think it would be possible to always have the dht_storage_constructor field? In general, it makes things a lot simpler if the ABI is not affected by build options like these